### PR TITLE
Remove sleep(1) from spawner

### LINKED ISSUE #160: controller_manager: spawner blocks indefinitely if manager shuts down first
Currently when bringing up a spawner and a controller manager, the spawner blocks on shutdown if the controller manager is shut down first. This wastes a lot of time during testing when bringing up the spawner and manager at the same time in simulation.

Instead, there should be a shorter timeout or a more reliable way of determining if the controller manager is still running.

### PR DESCRIPTION
**Pull request by bmagyar**
Fixes #160

Secretly hoping this saves us some time on Travis as well...

### CODE REVIEW

**Comment by mathias-luedtke** [comment]
I would even remove the `wait_for_service` calls.
IMHO if a controller was started (via a service!) there is not reason to wait for the service.
[rospy API](http://web.archive.org/web/20170223112144/http://docs.ros.org:80/kinetic/api/rospy/html/rospy.impl.tcpros_service-module.html#wait_for_service):
> NOTE: timeout=0 is invalid as wait_for_service actually contacts the service, so non-blocking behavior is not possible. For timeout=0 uses cases, just call the service without waiting

**Review by mathias-luedtke — CHANGES_REQUESTED** [review]
> I would even remove the wait_for_service calls.

**Comment by mathias-luedtke** [comment]
:+1: for the changes.
We should remove `shutdown_timeout` as well.
And display a deprecation warning if it was set in the command line.

**Comment by bmagyar** [comment]
I've done some other small updates on the code, should all be minor things but please give it a look, I've tested with `ros_controllers` but we should have separate tests for these guys hence #287

File: controller_manager/scripts/spawner [hunk 1]
```diff
@@ -85,21 +82,20 @@ def parse_args(args=None):
     parser.add_argument('--timeout', metavar='T', type=float, default=30,
                         help='how long to wait for controller_manager services when starting up [s] (default: 30)')
     parser.add_argument('--shutdown-timeout', metavar='T', type=float, default=30,
-                        help='how long to wait for controller_manager services when shutting down [s] (default: 30)')
```
Comment (mathias-luedtke) [inline controller_manager/scripts/spawner]: I would remove the default in the line before

File: controller_manager/scripts/spawner [hunk 2]
```diff
@@ -85,21 +82,20 @@ def parse_args(args=None):
...
 
     args = parse_args(rospy.myargv()[1:])
 
     wait_for_topic = args.wait_for
     autostart = 1 if not args.stopped else 0
     robot_namespace = args.namespace or ""
     timeout = args.timeout
-    shutdown_timeout = args.shutdown_timeout
 
     rospy.init_node('spawner', anonymous=True)
```
Comment (mathias-luedtke) [inline controller_manager/scripts/spawner:101]: And print a deprecation warning if `args.shutdown_timeout` was set (i.e. is not None).

**Comment by bmagyar** [comment]
Addressed the comments above

File: controller_manager/scripts/spawner [hunk 3]
```diff
@@ -88,22 +81,23 @@ def parse_args(args=None):
                         help='namespace of the controller_manager services')
     parser.add_argument('--timeout', metavar='T', type=float, default=30,
                         help='how long to wait for controller_manager services when starting up [s] (default: 30)')
-    parser.add_argument('--shutdown-timeout', metavar='T', type=float, default=30,
```
Comment (mathias-luedtke) [inline controller_manager/scripts/spawner:91]: I guess at least ` metavar='T', type=float` should be kept. 
Haven't tested it yet myself.

Comment (bmagyar) [reply controller_manager/scripts/spawner:91]: I've tested it, works fine with the defaults.

[metavar](https://docs.python.org/3/library/argparse.html#metavar):  "By default, ArgumentParser objects use the dest value as the “name” of each object. "
[type](https://docs.python.org/3/library/argparse.html#type): "By default, ArgumentParser objects read command-line arguments in as simple strings."

File: controller_manager/scripts/spawner (hunk 2 above)
Comment (bmagyar) [reply controller_manager/scripts/spawner:101]: Done

**Comment by bmagyar** [comment]
Feel free to merge if you're happy with it

**Comment by bmagyar** [comment]
ping @ipa-mdl

**Review by mathias-luedtke — APPROVED** [review]

**Pull request merged**